src/library/src/ubsan_options.cpp

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2020 Roc authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include "roc_core/attributes.h"
+
+#define SANITIZER_HOOK_ATTRIBUTE                                                         \
+    extern "C" ROC_ATTR_NO_SANITIZE_ADDR ROC_ATTR_NO_SANITIZE_UB                         \
+        __attribute__((visibility("default"))) __attribute__((used))
+
+SANITIZER_HOOK_ATTRIBUTE
+const char* __ubsan_default_options();
+
+SANITIZER_HOOK_ATTRIBUTE
+const char* __ubsan_default_options() {
+    return "print_stacktrace=1";
+}
```

Comment:
It seems that the only attribute really needed here is `__attribute__((visibility("default")))`. Others can be removed. Or maybe I'm missing something? I've tested it on recent gcc and clang on my machine.

Also, let's move definition of `__attribute__((visibility("default")))` to `roc_core/attributes.h`.

After doing this, I think we can remove `SANITIZER_HOOK_ATTRIBUTE` macro. Since we don't need `ROC_ATTR_NO_SANITIZE_ADDR`, we can remove it too.

Finally, let's rename this file to more generic `sanitizer_options.cpp` (we may add options to other sanitizers in future) and move it from ` src/library/src` to `src` (it's not specific to libroc and event not used in libroc at all).

Reply:
> It seems that the only attribute really needed here is `__attribute__((visibility("default")))`. Others can be removed. Or maybe I'm missing something? I've tested it on recent gcc and clang on my machine.

I'm not much familiar with this. But please refer to comment(line 20-23) [here](https://gn.googlesource.com/gn/+/e375226345b2be2bfc8c4549702ee3e437b5c136/build/sanitizers/sanitizer_options.cc).

Reply:
I see. So the reasoning on no_sanitize is to ensure that __ubsan_default_options() is not instrumented since it's called from sanitizer (though I also don't understand the exact problem). Then I'd suggest to add only ROC_ATTR_NO_SANITIZE_UB (since I supposed it's called only from UBSAN).

As for `__attribute__((used))`, I suppose it was needed to suppress a warning, but in our case it seems there is no warning.